### PR TITLE
Fix globalconstraints in a multiperiod context

### PR DESCRIPTION
**Pull request by coroa**
And some pandas deprecations.

### CODE REVIEW

File: pypsa/optimization/global_constraints.py [hunk 1]
```diff
@@ -389,12 +391,8 @@ def define_transmission_volume_expansion_limit(n, sns):
     m = n.model
     glcs = n.global_constraints.query("type == 'transmission_volume_expansion_limit'")
 
-    def substr(s):
-        return re.sub("[\\[\\]\\(\\)]", "", s)
-
-    lhs = []
     for name, glc in glcs.iterrows():
-        car = [substr(c.strip()) for c in glc.carrier_attribute.split(",")]
```
Comment (fneum) [inline pypsa/optimization/global_constraints.py:397]: I think `car` is still needed. It's used a few lines down in a query string.

**Review by fneum — CHANGES_REQUESTED** [review]
LGTM, almost.

One comment and a release note, then we can merge.

File: pypsa/optimization/global_constraints.py (hunk 1 above)
Comment (coroa) [reply pypsa/optimization/global_constraints.py:397]: Ah, yes, damn. Never trust VS Code blindly. The inquery string usage was lost on it.

**Comment by coroa** [comment]
Thanks for the review. Comments should be addressed. Please feel free to squash!

**Comment by coroa** [comment]
> Ah, yes, damn. Never trust VS Code blindly. The query string usage was lost on it.

**Pull request merged**